### PR TITLE
Check for WP_Error in wp_remote_get; fixes #2386

### LINKED ISSUE #2386: ISBAT PHP Fatal error:  Uncaught Error: Cannot use object of type WP_Error
### Describe the bug:
Fatal error being thrown:
```
PHP Fatal error:  Uncaught Error: Cannot use object of type WP_Error as array in /srv/users/.../plugins/coblocks/src/blocks/gist/index.php:50
Stack trace:
#0 /srv/users/.../wp-includes/class-wp-embed.php(156): coblocks_block_gist_handler(Array, Array, 'https://gist.gi...', Array)
#1 /srv/users/.../wp-includes/class-wp-embed.php(219): WP_Embed->get_embed_handler_html(Array, 'https://gist.gi...')
#2 /srv/users/.../wp-includes/class-wp-embed.php(463): WP_Embed->shortcode(Array, 'https://gist.gi...')
#3 [internal function]: WP_Embed->autoembed_callback(Array)
#4 /srv/users/.../wp-includes/class-wp-embed.php(445): preg_replace_callback('|^(\\s*)(https?:...', Array, '<!-- wp:paragra...')
#5 /srv/users/...wp-includes/class-wp-hook.php(307): WP_Embed->autoembed('<!-- wp:paragra...')
#6 /srv/users/.../wp-includes/plugin.php( in /srv/users/.../wp-content/plugins/coblocks/src/blocks/gist/index.php on line 50
```

### To reproduce:
Log into WP Admin, note all dashboard widgets missing.

### Expected behavior:
No error, or blocker in admin index dashboard page.

### Isolating the problem:

- [ ] This bug happens with no other plugins activated
- [ ] This bug happens with a default WordPress theme active
- [ ] This bug happens **without** the Gutenberg plugin active
- [ ] I can reproduce this bug consistently using the steps above

### WordPress version:
5.9.3

### PHP version:
7.4.29

### PR DESCRIPTION
**Pull request by thefrosty**
### Description

Fixes #2386 

### Checklist:
- [ ] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

**Comment by olafleur-godaddy** [comment]
The pipeline does not seem to run our your PR (maybe a config issue), but I can see that the PHP tests and linting are failing (ex : https://app.circleci.com/pipelines/github/godaddy-wordpress/coblocks/9503/workflows/1e53351b-47d1-4a51-812f-4e7d3f87773f ) 

Please make sure that the `yarn lint:php` and `yarn test:php` commands are not failing.

Thank you!

**Comment by olafleur-godaddy** [comment]
The pipeline does not seem to run our your PR (maybe a config issue), but I can see that the PHP tests and linting are failing (ex : https://app.circleci.com/pipelines/github/godaddy-wordpress/coblocks/9503/workflows/1e53351b-47d1-4a51-812f-4e7d3f87773f ) 

Please make sure that the `yarn lint:php` and `yarn test:php` commands are not failing.

Thank you!

**Review by olafleur-godaddy — CHANGES_REQUESTED** [review]
The pipeline does not seem to run our your PR (maybe a config issue), but I can see that the PHP tests and linting are failing (ex : https://app.circleci.com/pipelines/github/godaddy-wordpress/coblocks/9503/workflows/1e53351b-47d1-4a51-812f-4e7d3f87773f )

Please make sure that the `yarn lint:php` and `yarn test:php` commands are not failing.

Thank you!

File: src/blocks/gist/index.php [hunk 1]
```diff
@@ -45,18 +45,20 @@ function coblocks_block_gist_handler( $matches ) {
...
 	);
 
-	// JSON decode the response body.
-	// Then grab files from JSON object.
-	$body  = json_decode( $api_resp['body'], true );
-	$files = empty( $body['files'] ) ? array() : $body['files'];
+	$resp_body = wp_remote_retrieve_body( $api_resp );
+	$result = json_decode( $resp_body, true );
+	
+	if ( !is_array( $result ) || empty( $result['files'] ) || is_wp_error( $result ) {
```
Comment (thefrosty) [inline src/blocks/gist/index.php:51]:
```suggestion
	if ( ! is_array( $result ) || empty( $result['files'] ) || is_wp_error( $result ) ) {
```

**Review by AnthonyLedesma — APPROVED** [review]
Excellent change here. We appreciate the effort to get this PR in and fixed up.

**Comment by olafleur-godaddy** [comment]
Yes, thanks for your contribution!

**Comment by olafleur-godaddy** [comment]
I just added you in the contributors list : https://github.com/godaddy-wordpress/coblocks/blob/master/CONTRIBUTORS.md

**Pull request merged**